docs/source/topics/data_types.rst

```diff
@@ -305,14 +305,51 @@ This class can be initialized via the **absolute** path to the file you want to
 
   In [2]: single_file = SinglefileData('/absolute/path/to/file')
 
+When storing the node, the filename is stored in the database and the file itself is copied to the repository.
 The contents of the file in string format can be obtained using the :py:meth:`~aiida.orm.nodes.data.singlefile.SinglefileData.get_content()` method:
 
 .. code-block:: ipython
 
   In [3]: single_file.get_content()
   Out[3]: 'The file content'
 
-When storing the node, the filename is stored in the database and the file itself is copied to the repository.
+For large files, reading the entire content into memory using :py:meth:`~aiida.orm.nodes.repository.NodeRepository.get_object_content()` may not be desirable.
+Instead, a file-like handle can be opened to a file in the repository which can be used to read the content as a stream.
+This can be useful, for example, to copy a large file from the repository to a file on disk, without loading it entirely into memory:
+
+.. code-block:: ipython
+
+    In [4]: import shutil
+            with single_file.open(mode='rb') as source:
+                with open('copy.txt', mode='wb') as target:
+                    shutil.copyfileobj(source, target)
+
+.. note:: To guarantee the file is copied over identically (and there are no encoding issues), the files are opened in "binary" mode by including the ``b`` character in the ``mode`` argument.
+
+For efficiency reasons, the repository interface only provides access to object content through file-like objects or strings.
+However, for certain use-cases, the object content _needs_ to be made available as a file on the local file system.
+For example, the ``numpy.loadtxt`` method only accepts a filepath, and no file-like objects.
+In this case, the content of the file can be made available on the local file system using the :py:meth:`~aiida.orm.nodes.repository.NodeRepository.as_path()` context manager:
+
+.. code-block:: ipython
+
+    In [5]: with single_file.as_path() as filepath:
+                numpy.loadtxt(filepath)
+
```

Comment:
Since you mention above generally `shutil.copyfileobj`, shouldn't we have an example of that here as well?

Reply:
But `shutil.copyfileobj` takes a file handle, not a filepath. Of course I could then do
```python
with single_file.as_path() as filepath:
    with filepath.open() as source:
        with open('output.txt', 'wb') as target:
            shutil.copyfileobj(source, target)
```
That would be defeating the point. Or did you mean another method _like_ `shutil.copyfileobj` but that uses filepaths?